src/content-script/in-page.ts

```diff
@@ -31,6 +33,9 @@ if (!walletChannelId) {
 const broadcastChannel = new BroadcastChannel(walletChannelId);
 const connection = new Connection(broadcastChannel);
 const provider = new EthereumProvider(connection);
+const zerionSolana = new ZerionSolana(connection);
+initialize(zerionSolana);
+Object.assign(provider, { solana: zerionSolana });
```

Comment:
May be it is worth placing solana and ethereum in one level?
`provider: { solana, ethereum }`

Reply:
`provider` is the value being assigned to `window.ethereum`, so its shape cannot be changed

Assigning `solana` to it as I do here is mostly a helper for debugging. Perhaps it's not the most correct thing to do